pilot/pkg/proxy/envoy/discovery.go

```diff
@@ -112,10 +112,21 @@ var (
 
 var (
 	// Variables associated with clear cache squashing.
-	lastClearCache     time.Time
+
+	// lastClearCache is the time we last pushed
+	lastClearCache time.Time
+
+	// lastClearCacheEvent is the time of the last config event
+	lastClearCacheEvent time.Time
+
+	// clearCacheTimerSet is true if we are in squash mode, and a timer is already set
 	clearCacheTimerSet bool
-	clearCacheMutex    sync.Mutex
-	clearCacheTime     = 1
+
+	clearCacheMutex sync.Mutex
```

Comment:
Put the mutex at the top of this `var` block since it protects all of the variables here (mirroring the pattern that a mutex struct field should come before the other struct fields it guards).